libs/header/p2p/exchange.go

```diff
@@ -220,13 +229,21 @@ func (ex *Exchange[H]) performRequest(
 		return make([]H, 0), nil
 	}
 
-	if len(ex.trustedPeers) == 0 {
-		return nil, fmt.Errorf("no trusted peers")
+	for {
+		//nolint:gosec // G404: Use of weak random number generator
+		idx := rand.Intn(len(ex.trustedPeers))
+		ctx, cancel := context.WithTimeout(ctx, ex.Params.TrustedPeersRequestTimeout)
+
+		h, err := ex.request(ctx, ex.trustedPeers[idx], req)
+		cancel()
+		switch err {
+		default:
+			log.Debug(err)
```

Comment:
```suggestion
			log.Debugw("request failed", "peer", tpeer.String(), "err", err)
```

Reply:
All the errors are wrapped in `request`, so doing this duplicate info mostly make the whole log uglier. I tried it originally